python/lsst/analysis/tools/actions/plot/focalPlanePlot.py

```diff
@@ -230,3 +232,253 @@ def makePlot(
+            corners = np.array(corners)
+
+            # See if the plot bounding box needs to be extended:
+            for corner in corners:
+                if corner[0] < plotLimit_x[0]:
+                    plotLimit_x[0] = corner[0]
+                if corner[0] > plotLimit_x[1]:
+                    plotLimit_x[1] = corner[0]
+                if corner[1] < plotLimit_y[0]:
+                    plotLimit_y[0] = corner[1]
+                if corner[1] > plotLimit_y[1]:
+                    plotLimit_y[1] = corner[1]
+
+            # U/V coordinates represent focal plane locations.
+            minU, minV = corners.min(axis=0)
+            maxU, maxV = corners.max(axis=0)
+
+            # X/Y coordinates represent detector internal coordinates.
+            # Detector extent in detector coordinates
+            minX, minY = detector.getBBox().getMin()
+            maxX, maxY = detector.getBBox().getMax()
+
+            if self.level.lower() == "detector":
+                detectorInd = data["detector"] == detectorId
+
+                # This does the appropriate statistic for this
+                # detector's data.
+                statistic, _, _ = binned_statistic_dd(
+                    [focalPlane_x[detectorInd], focalPlane_y[detectorInd]],
+                    data["z"][detectorInd],
+                    statistic=self.statistic,
+                    bins=[1, 1],
+                )
+                patches.append(Polygon(corners, True))
+                values.append(statistic.ravel()[0])
+            else:
+                # It's at amplifier level.  This uses the focal
+                # plane position of the corners of the detector to
+                # generate corners for the individual amplifier
+                # segments.
+                rotation = detector.getOrientation().getNQuarter()  # N * 90 degrees.
+                alpha, beta = np.cos(rotation * np.pi / 2.0), np.sin(rotation * np.pi / 2.0)
+
+                # Calculate the rotation matrix between X/Y and U/V
+                # coordinates.
+                scaleUX = alpha * (maxU - minU) / (maxX - minX)
+                scaleVX = beta * (maxV - minV) / (maxX - minX)
+                scaleVY = alpha * (maxV - minV) / (maxY - minY)
+                scaleUY = beta * (maxU - minU) / (maxY - minY)
+
+                # After the rotation, some of the corners may have
+                # negative offsets.  This corresponds to corners that
+                # reference the maximum edges of the box in U/V
+                # coordinates.
+                baseU = minU if rotation % 4 in (0, 1) else maxU
+                baseV = maxV if rotation % 4 in (2, 3) else minV
+
+                for amplifier in detector:
+                    ampName = amplifier.getName()
+                    detectorInd = data["detector"] == detectorId
+                    ampInd = data["x"] == ampName
+                    detectorInd &= ampInd
```

Comment:
Personally, I think it would be clearer to reverse this to `ampInd &= detectorInd` so that you are using `ampInd` later.